File: Solution1/BootStrapSample/js/Qna.js
//function GetContents() {
//    document.getElementById('editorcontents').innerHTML = CKEDITOR.instances.askEditor.getData();
//}

// 질문목록 조회
function GetAskList() {

    var searchType = '';
    var searchText = document.getElementById('txtAskSearchText').value;

    var data = { 'searchType': searchType, 'searchText': searchText };
    DoAjaxCall('GetAskList', '', 'GetAskListCallBack', data);
}

// 질문목록 조회 콜백
function GetAskListCallBack(data) {
    var str = "<table border='1' style='width:500px;'>"
    str += "<th>좋은질문</th><th>사용자</th><th>제목</th><th>조회수</th><th>답변수</th><th>등록일</th>"
    for (var i = 0; i < data.length; i++) {
        str += "<tr>";
        str += "<td>" + data[i].GoodAskCount + "</td>";
        str += "<td>" + data[i].UserNickName + "</td>";
        str += "<td><a href='AskDetail.aspx?seq=" + data[i].AskSeq + "'>" + data[i].AskTitle + "</a></td>";
        str += "<td>" + data[i].AskCount + "</td>";
        str += "<td>" + data[i].AnswerCount + "</td>";
        str += "<td>" + data[i].AskRegDate + "</td></tr>";
    }
    str += "</table>";

    $("#divAskList").html(str);
}

// 질문하기
function SetAsk() {
    if (window.sessionStorage['LoginKey']) {
        var userSeq = sessionStorage['LoginKey'];
        var title = document.getElementById('txtAskTitle').value;
        var ask = htmlEscape(CKEDITOR.instances.askEditor.getData());
        var data = { 'userSeq': userSeq, 'title': title, 'ask': ask };
        DoAjaxCall('SetAsk', '', 'SetAskCallBack', data);
    }
}

// 질문하기 콜백
function SetAskCallBack(data) {
    alert('정상적으로 저장되었습니다.');
    window.location = 'Ask.aspx';
}

// 질문상세 조회
function GetAskDetail() {
    var seq = $.getUrlVar('seq');
    var data = { 'seq': seq };
    DoAjaxCall('GetAskDetail', '', 'GetAskDetailCallBack', data);
}

// 질문상세 콜백
function GetAskDetailCallBack(data) {
    //제목
    var strTitle = "<input type='hidden' id='hidAskSeq' value='" + data[0].AskSeq + "' />";
    strTitle += "<div class='page-header'><h2>" + data[0].AskTitle + "</h2></div>";
    $("#divAskDetailTitle").html(strTitle);

    //상세내용
    var strDoc = data[0].AskDoc;
    document.getElementById('divAskDetailDoc').innerHTML = "<p>" + htmlUnescape(strDoc) + "</p>";

    // 답변
    var strAnswer = '';
    for (var i = 0; i < data[0].Answers.length; i++) {
        strAnswer += "<table class='table table-bordered'>";
        strAnswer += "<thead><tr><th colspan='2'>사용자:" + data[0].Answers[i].UserNickName + "</th></tr></thead>";
        strAnswer += "<tbody><tr id='class-summary'><td><p class='class-scope'><a onclick='GoodAnswer(" + data[0].Answers[i].AnswerSeq + ");'>좋은답변</a><span id=span" + data[0].Answers[i].AnswerSeq + ">" + data[0].Answers[i].GoodAnswerCount + "</span></p></td>"
        strAnswer += "<td><p class='class-desc'>" + data[0].Answers[i].Answer + "</p></td></tbody></table>";
    }
    if (strAnswer.length > 0)
        document.getElementById('divAnswer').innerHTML = htmlUnescape(strAnswer);
}

// 좋은답변
function GoodAnswer(seq) {
    var userSeq;
    if (window.sessionStorage['LoginKey']) {
        userSeq = sessionStorage['LoginKey'];
    }
    else {
        userSeq = -1;
    }
    var data = { 'seq': seq, 'userSeq': userSeq };
    DoAjaxCall('GoodAnswer', '', 'GoodAnswerCallBack', data);
}

// 좋은답변 콜백
function GoodAnswerCallBack(data) {
    var answer = document.getElementById('span' + data[0]);
    answer.textContent = data[1];
}

// 답변하기
function SetAnswer() {
    var askSeq = document.getElementById('hidAskSeq').value;
    var answer = htmlEscape(CKEDITOR.instances.answerEditor.getData());
    //var userSeq = sessionStorage['UserSeq'];
    var userSeq = '1'
    var data = { 'askSeq': askSeq, 'answer': answer, 'userSeq': userSeq };
    DoAjaxCall('SetAnswer', '', 'SetAnswerCallBack', data);
}

// 답변하기 콜백
function SetAnswerCallBack(data) {
    alert('정상적으로 저장되었습니다.');
    window.location = window.location;
}

File: trunk/Solution1/MAQNA.DAL/ViewClass/QNA.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace MAQNA.DAL.ViewClass
{
    public class AskList
    {
        public string AskSeq { get; set; }
        public string UsersSeq { get; set; }
        public string AskTitle { get; set; }
        public string AskCount { get; set; }
        public string AskRegDate { get; set; }
        public string AnswerCount { get; set; }
        public string GoodAskCount { get; set; }
        public string UserNickName { get; set; }

        public static List<AskList> ConvertList(DataSet ds)
        {
            List<AskList> list = new List<AskList>();
            foreach (DataRow row in ds.Tables[0].Rows)
            {
                AskList data = new AskList();
                data.AskSeq = row["AskSeq"].ToString();
                data.UsersSeq = row["UsersSeq"].ToString();
                data.AskTitle = row["AskTitle"].ToString();
                data.AskCount = row["AskCount"].ToString();
                data.AnswerCount = row["AnswerCount"].ToString();
                data.GoodAskCount = row["GoodAskCount"].ToString();
                data.AskRegDate = row["AskRegDate"].ToString();
                data.UserNickName = row["UserNickName"].ToString();
                list.Add(data);
            }

            return list;
        }
    }

    public class AskDetail
    {
        public string AskSeq { get; set; }
        public string AskTitle { get; set; }
        public string AskDoc { get; set; }
        public string AskCount { get; set; }
        public string UsersSeq { get; set; }
        public string UserNickName { get; set; }
        public List<AnswerList> Answers { get; set; }

        public static List<AskDetail> ConvertList(DataSet ds)
        {
            List<AskDetail> list = new List<AskDetail>();
            foreach (DataRow row in ds.Tables[0].Rows)
            {
                AskDetail data = new AskDetail();
                data.AskSeq = row["AskSeq"].ToString();
                data.UsersSeq = row["UsersSeq"].ToString();
                data.AskTitle = row["AskTitle"].ToString();
                data.AskDoc = row["AskDoc"].ToString();
                data.AskCount = row["AskCount"].ToString();
                data.UserNickName = row["UserNickName"].ToString();
                data.Answers = new List<AnswerList>();
                foreach (DataRow row1 in ds.Tables[1].Rows)
                {
                    AnswerList data1 = new AnswerList();
                    data1.AnswerSeq = row1["AnswerSeq"].ToString();
                    data1.Answer = row1["Answer"].ToString();
                    data1.UsersSeq = row1["UsersSeq"].ToString();
                    data1.UserNickName = row1["UserNickName"].ToString();
                    data1.GoodAnswerCount = row1["GoodAnswerCount"].ToString();
                    data.Answers.Add(data1);
                }

                list.Add(data);
            }

            return list;
        }

        public class AnswerList
        {
            public string AnswerSeq { get; set; }
            public string Answer { get; set; }
            public string UsersSeq { get; set; }
            public string UserNickName { get; set; }
            public string GoodAnswerCount { get; set; }
        }
    }

    public class RegAskDetail
    {
        public string AskSeq { get; set; }
        public string AskTitle { get; set; }
        public string AskDoc { get; set; }
        public string AskCount { get; set; }
        public string UsersSeq { get; set; }
        public string UserNickName { get; set; }

        public static List<RegAskDetail> ConvertList(DataSet ds)
        {
            List<RegAskDetail> list = new List<RegAskDetail>();
            foreach (DataRow row in ds.Tables[0].Rows)
            {
                RegAskDetail data = new RegAskDetail();
                data.AskSeq = row["AskSeq"].ToString();
                data.UsersSeq = row["UsersSeq"].ToString();
                data.AskTitle = row["AskTitle"].ToString();
                data.AskDoc = row["AskDoc"].ToString();
                data.AskCount = row["AskCount"].ToString();
                data.UserNickName = row["UserNickName"].ToString();
                list.Add(data);
            }

            return list;
        }
    }
}
